definitions/types/azure-security-iam.json

```diff
@@ -1,24 +1,34 @@
 {
   "$schema": "http://json-schema.org/draft-07/schema",
   "type": "object",
-  "title": "Azure Security IAM Information",
-  "description": "Informs downstream services of IAM roles required for operation",
+  "title": "IAM",
+  "description": "IAM Roles And Conditions",
   "additionalProperties": false,
-  "required": [
-    "roles",
-    "scope"
-  ],
-  "properties": {
-    "roles": {
-      "title": "List of Azure IAM roles",
-      "type": "array",
-      "items": {
-        "type": "string"
+  "patternProperties": {
+    "^[a-z]+$": {
+      "type": "object",
+      "required": [
+        "role",
+        "scope"
+      ],
+      "properties": {
+        "role": {
+          "title": "Role",
+          "description": "Azure Role",
+          "pattern": "^[a-zA-Z ]+$",
+          "message": {
+            "pattern": "Must be a valid Azure Role (uppercase, lowercase letters and spaces)"
+          },
+          "examples": [
+            "Storage Blob Data Reader"
+          ]
+        },
+        "scope": {
+          "title": "Scope",
+          "description": "Azure IAM Scope",
+          "type": "string"
```

Comment:
Is there any patterns to the scope strings we could regex on?

Reply:
Yeah, I think scope can be a subscription, resource group, or resource id.  I suppose its a question of whether we make a generic `^[a-zA-Z0-9-]\/+` sort of pattern that fits all 3, or if we have separate types for all 3.

Here is what a resource id looks like, and you can infer what the other two would like like from it:
`"/subscriptions/12345678-1234-1234-abcd-1234567890ab/resourceGroups/resource-group-name/providers/Microsoft.Network/virtualNetworks/network-name"`

Reply:
something similar to this: `\/subscriptions\/[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}\/resourceGroups\/[0-9A-Za-z-]+\/providers\/Microsoft\.[a-zA-Z]+\/[a-zA-Z]+\/[0-9A-Za-z-]+`

Reply:
@mclacore and I will circle back to this when we add types for subscription, resource-group and resource-id